### PR TITLE
Add CuArray convert adjoint

### PR DESCRIPTION
**Pull request by dhairyagandhi96**
Fixes FluxML/Flux.jl#947

### CODE REVIEW

File: src/lib/broadcast.jl [hunk 1]
```diff
@@ -187,4 +187,8 @@ end
     sum(xs, dims = dims), Δ -> (placeholder .= Δ,)
   end
 
+  @adjoint function Base.convert(::Type{T}, xs::Array{K,N})  where {T<:CuArrays.CuArray, K, N}
```
Comment (MikeInnes) [inline src/lib/broadcast.jl]: The `K, N` could be removed here, right? Otherwise LGTM.

Comment (dhairyagandhi96) [reply src/lib/broadcast.jl]: Should be fixed now, apologies for the delay

**Pull request merged**